test-app/tests/integration/components/file-input-field-test.gts

```diff
@@ -0,0 +1,326 @@
+
+    assert.verifySteps([]);
+
+    const file = createFile(['Upload file sample'], {
+      name: 'sample.txt',
+      type: 'text/plain',
+    });
+
+
+    await triggerEvent('[data-file-input-field]', 'change', { files: [file] });
+
+    assert.verifySteps(['realOnChange']);
+
+    assert.dom('[data-files] [data-file-name]').hasText('sample.txt');
+
+    // currently: rounding to the nearest KB
+    // this file is very small so result is 0 KB
+    assert.dom('[data-files] [data-file-size]').hasText('0 KB');
+  });
+
+  test('it deletes a file', async function(assert) {
+    assert.expect(8)
+
+    class Context {
+      @tracked currentFiles: File[] = [];
+    }
+
+    let ctx = new Context();
+
+    const realOnChange = (files: File[]) => {
+      ctx.currentFiles = files;
+    }
+    
+    const realOnDelete = (file: File) => {
+      assert.ok(file, 'Expected a single file to exist in the target');
+      assert.ok(file instanceof File, 'Expected a single file to be a File object');
+      assert.strictEqual(file.name, 'sample.txt', 'Expected a single file to have a name');
+      assert.strictEqual(file.size, 18, 'Expected a single file to have a size');
+      assert.step('realOnDelete');
+      // the entire array gets replaced here, so no used of trackedArray
+      ctx.currentFiles = ctx.currentFiles.filter(currentFile => currentFile !== file); 
+    }
+
+    await render(<template>
+      <FileInputField
+        @label="Label"
+        @trigger="Select Files"
+        @onChange={{realOnChange}}
+        @onDelete={{realOnDelete}}
+        @files={{ctx.currentFiles}}
+        data-file-input-field
+      />
+    </template>);
+
+    assert.verifySteps([]);
+
+    const file = createFile(['Upload file sample'], {
+      name: 'sample.txt',
+      type: 'text/plain',
+    });
+
+    await triggerEvent('[data-file-input-field]', 'change', { files: [file] });
+
+    await triggerEvent('button', 'click');
+    
+    assert.verifySteps(['realOnDelete'])
+
+    assert.dom('ul').doesNotExist();
+  });
+
+  test('it applies the provided @rootTestSelector to the data-root-field attribute', async function (assert) {
+    await render(<template>
+      <FileInputField
+        @label="Label"
+        @trigger="Select Files"
+        @onChange={{onChange}}
+        @onDelete={{onDelete}}
+        @rootTestSelector="selector"
+        data-file-input-field
+      />
+    </template>);
+
+    assert.dom('[data-root-field="selector"]').exists();
+  });
+
+  test('it throws an assertion error if no @label is provided', async function (assert) {
+    assert.expect(1);
+
+    setupOnerror((e: Error) => {
+      assert.ok(
+        e.message.includes('A "@label" argument is required'),
+        'Expected assertion error message'
+      );
+    });
+
+    await render(<template>
+      {{! @glint-expect-error: we are not providing @label, so this is expected }}
+      <FileInputField @onChange={{onChange}} />
+    </template>);
+  });
```

Comment:
It looks like we require a few other arguments as well - should we write tests for these too?

```
trigger
onChange
onDelete
```

Reply:
Should be in now.